src/sentry/snuba/metrics/helpers.py

```diff
@@ -516,67 +533,64 @@ def __init__(
         self._intervals = intervals
         self._results = results
 
-        self._ops_by_metric = ops_by_metric = {}
-        for op, metric in query_definition.fields.values():
-            ops_by_metric.setdefault(metric, []).append(op)
-
         self._timestamp_index = {timestamp: index for index, timestamp in enumerate(intervals)}
 
     def _parse_tag(self, tag_string: str) -> str:
         tag_key = int(tag_string.replace("tags[", "").replace("]", ""))
         return reverse_resolve(tag_key)
 
-    def _extract_data(self, entity, data, groups):
+    def _extract_data(self, data, groups):
         tags = tuple(
             (key, data[key])
             for key in sorted(data.keys())
             if (key.startswith("tags[") or key in ALLOWED_GROUPBY_COLUMNS)
         )
 
-        metric_name = reverse_resolve(data["metric_id"])
-        ops = self._ops_by_metric[metric_name]
-
         tag_data = groups.setdefault(
             tags,
-            {
-                "totals": {},
-            },
+            {"totals": {}, "series": {}},
         )
 
-        timestamp = data.pop(TS_COL_GROUP, None)
-        if timestamp is not None:
-            timestamp = parse_snuba_datetime(timestamp)
-
-        for op in ops:
-            key = f"{op}({metric_name})"
-
-            value = data[op]
-            if op in _OPERATIONS_PERCENTILES:
-                value = value[0]
-
-            # If this is time series data, add it to the appropriate series.
-            # Else, add to totals
-            if timestamp is None:
-                tag_data["totals"][key] = finite_or_none(value)
-            else:
-                series = tag_data.setdefault("series", {}).setdefault(
-                    key, len(self._intervals) * [_DEFAULT_AGGREGATES[op]]
-                )
-                series_index = self._timestamp_index[timestamp]
-                series[series_index] = finite_or_none(value)
+        bucketed_time = data.pop(TS_COL_GROUP, None)
+        if bucketed_time is not None:
+            bucketed_time = parse_snuba_datetime(bucketed_time)
+
+        for op, metric_name in self._query_definition.fields.values():
+            try:
+                key = f"{op}({metric_name})"
+                value = data[key]
+                if op in _OPERATIONS_PERCENTILES:
+                    value = value[0]
+                cleaned_value = finite_or_none(value)
+            except KeyError:
+                continue
```

Comment:
Should we add a comment here about why this `KeyError` is expected?

Reply:
all good feedback .. thanks!